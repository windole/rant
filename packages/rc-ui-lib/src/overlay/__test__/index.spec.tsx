import React from 'react';
import { mount } from 'enzyme';
import { Overlay } from '..';

describe('Overlay', () => {
  it('should change z-index when using z-index prop', () => {
    const wrapper = mount(<Overlay visible zIndex={99} />);
    const overlay = wrapper.find('.rc-overlay');
    expect(overlay.getDOMNode().style.zIndex).toEqual('99');
  });

  it('should allow to custom class name with class-name prop', () => {
    const wrapper = mount(<Overlay visible className="foo" />);
    const overlay = wrapper.find('.rc-overlay');
    expect(overlay.contains('foo'));
  });

  it('should allow to custom style with custom-style prop', () => {
    const wrapper = mount(<Overlay visible customStyle={{ backgroundColor: 'red' }} />);
    const overlay = wrapper.find('.rc-overlay');
    expect(overlay.getDOMNode().style.backgroundColor).toEqual('red');
  });

  it('should change animation duration when using duration prop', () => {
    const wrapper = mount(<Overlay visible duration={100} />);
    const overlay = wrapper.find('.rc-overlay');
    expect(overlay.getDOMNode().style.animationDuration).toEqual('100ms');
  });

  it('should render default slot correctly', () => {
    const wrapper = mount(<Overlay>Custom Default</Overlay>);
    expect(wrapper).toMatchSnapshot();
  });
});
