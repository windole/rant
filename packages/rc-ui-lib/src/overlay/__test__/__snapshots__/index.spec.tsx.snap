// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render default slot correctly 1`] = `
<Overlay
  duration={300}
  lockScroll={true}
>
  <CSSTransition2
    classNames="rc-fade"
    mountOnEnter={true}
    nodeRef={
      Object {
        "current": null,
      }
    }
    timeout={300}
    unmountOnExit={true}
  >
    <Transition2
      appear={false}
      enter={true}
      exit={true}
      in={false}
      mountOnEnter={true}
      nodeRef={
        Object {
          "current": null,
        }
      }
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      timeout={300}
      unmountOnExit={true}
    />
  </CSSTransition2>
</Overlay>
`;
